packages/nextra-theme-blog/src/mdx-theme.tsx

```diff
@@ -27,25 +28,28 @@ const H1 = ({ children }: { children?: ReactNode }): ReactElement => {
     if (ref.current && opts.hasJsxInH1) {
       setShowHeading(true)
     }
-  }, [])
+  }, [ref, opts.hasJsxInH1])
```

Comment:
```suggestion
}, [opts.hasJsxInH1])
```

Reply:
Why this change? I have eslint warning about missing dependency, this will avoid adding `// eslint disable` comment and anyway we check `ref.current` inside `useEffect`

Reply:
seems this rule is dumb and omits warning only if `ref` is declared with hook `useRef`
https://github.com/facebook/react/blob/0b974418c9a56f6c560298560265dcf4b65784bc/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L226-L228